pkg/services/publicdashboards/publicdashboard.go

```diff
@@ -16,23 +16,25 @@ import (
 type Service interface {
 	AccessTokenExists(ctx context.Context, accessToken string) (bool, error)
 	BuildAnonymousUser(ctx context.Context, dashboard *models.Dashboard) (*user.SignedInUser, error)
-	GetPublicDashboard(ctx context.Context, accessToken string) (*PublicDashboard, *models.Dashboard, error)
 	GetDashboard(ctx context.Context, dashboardUid string) (*models.Dashboard, error)
 	GetMetricRequest(ctx context.Context, dashboard *models.Dashboard, publicDashboard *PublicDashboard, panelId int64, reqDTO PublicDashboardQueryDTO) (dtos.MetricRequest, error)
+	GetPublicDashboard(ctx context.Context, accessToken string) (*PublicDashboard, *models.Dashboard, error)
 	GetPublicDashboardConfig(ctx context.Context, orgId int64, dashboardUid string) (*PublicDashboard, error)
 	GetQueryDataResponse(ctx context.Context, skipCache bool, reqDTO PublicDashboardQueryDTO, panelId int64, accessToken string) (*backend.QueryDataResponse, error)
+	ListPublicDashboards(ctx context.Context, orgId int64) ([]PublicDashboardListResponse, error)
 	PublicDashboardEnabled(ctx context.Context, dashboardUid string) (bool, error)
 	SavePublicDashboardConfig(ctx context.Context, u *user.SignedInUser, dto *SavePublicDashboardConfigDTO) (*PublicDashboard, error)
 }
 
 //go:generate mockery --name Store --structname FakePublicDashboardStore --inpackage --filename public_dashboard_store_mock.go
 type Store interface {
 	AccessTokenExists(ctx context.Context, accessToken string) (bool, error)
-	GetDashboard(ctx context.Context, dashboardUid string) (*models.Dashboard, error)
 	GenerateNewPublicDashboardUid(ctx context.Context) (string, error)
+	GetDashboard(ctx context.Context, dashboardUid string) (*models.Dashboard, error)
```

Comment:
what does this do? Why would PublicDashboard Store have GetDashboard (feels like something that should be delegated to DashboardService (then to Dashboard store)

Reply:
This method was not introduced by this PR, it was just sorting methods alphabetically on the interface.

I agree in principle, however, we had a specific use case to query on uid only and wanted to limit dependency injection for another service. Additionally the query pattern imposed by the dashboard service is a bit of an anti pattern and as far as I knew, we were working to remove this.